modules/executive/api/fetchExecutives.ts

```diff
@@ -62,22 +62,21 @@ export async function getGithubExecutives(network: SupportedNetworks): Promise<C
   return sortedProposals;
 }
 
-async function getGithubExecutivesWithMKR(network: SupportedNetworks): Promise<CMSProposal[]> {
-  const proposals = await getGithubExecutives(network);
-
-  const mkrSupports = await Promise.all(
+async function getExecutivesSpellData(
+  network: SupportedNetworks,
+  proposals: CMSProposal[]
+): Promise<Proposal[]> {
+  const proposalsWithData = await Promise.all(
     proposals.map(async proposal => {
-      const mkrSupport = await getExecutiveMKRSupport(proposal.address, network);
```

Comment:
i removed before the analyzeSpell because it was taking too long to do it for all the executives. That's why i limited only to fetch the MKR support for sorting by MKR. 

In the client side is also using the analyze spell, so we should control in the FE that if the spell data comes from the response, don't fetch it again.

I did a sort by mkr on this deployment and the API endpoint times out, so we have to find another solution. Maybe using spock to get a query with all the MKR weights for all the executives is the answer.

Reply:
Did some digging into this but we're not storing MKR weights for executives in spock. I did some experimenting with the `JsonBatchRpcProvider` in ethers and it looks promising. We can fetch all the calls in analyze spell in a single request eliminating the need to do 9 calls in a row for each spell.